eva_submission/eload_ingestion.py

```diff
@@ -0,0 +1,111 @@
+import string
+import subprocess
+
+from bs4 import BeautifulSoup
+from ebi_eva_common_pyutils import command_utils
+from ebi_eva_common_pyutils.config import cfg
+from ebi_eva_common_pyutils.mongo_utils import get_mongo_connection_handle
+import requests
+
+from eva_submission.eload_submission import Eload
+
+
+def sanitize(s):
+    """Lowercases and removes punctuation from s, e.g. to make it a legal MongoDB identifier."""
+    return s.lower().translate(str.maketrans('', '', string.punctuation))
+
+
+class EloadIngestion(Eload):
+    config_section = 'ingestion'  # top-level config key
+    all_tasks = ['metadata_load', 'accession', 'variant_load']
+
+    def __init__(self, eload_number):
+        super().__init__(eload_number)
+        self.eload_cfg.set(self.config_section, 'ingestion_date', value=self.now)
+
+    def ingest(self, db_name=None, tasks=None):
+        # TODO assembly/taxonomy insertion script should be incorporated here
+        self.check_variant_db(db_name)
+
+        if not tasks:
+            tasks = self.all_tasks
+
+        if 'metadata_load' in tasks:
+            self.load_from_ena()
+        if 'accession' in tasks:
+            self.warning('Accessioning not yet supported, skipping.')
+        if 'variant_load' in tasks:
+            self.warning('Variant loading not yet supported, skipping.')
+
+    def get_db_name(self):
+        """
+        Constructs the expected database name in mongo, based on assembly info retrieved from ENA.
+        """
+        assm_accession = self.eload_cfg.query('submission', 'assembly_accession')
+        ena_url = f'https://www.ebi.ac.uk/ena/browser/api/xml/{assm_accession}'
+        try:  # catches any kind of request error, including non-20X status code
+            response = requests.get(ena_url)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            self.error(f"Couldn't get assembly info from ENA for accession {assm_accession}")
+            raise e
+
+        soup = BeautifulSoup(response.text, 'lxml')
+        sci_name_terms = soup.scientific_name.text.split()
+        sci_name = sci_name_terms[0][0] + ''.join(sci_name_terms[1:])
+        assembly_name = soup.find('name').text
```

Comment:
I see many <name> tags in the result. Guessing bs4 will only find the first occurrence which will correspond to the assembly? Nevertheless, it is better if we are more explicit about that...